streams-bootstrap-core/src/main/java/com/bakdata/kafka/util/ConsumerGroupClient.java

```diff
@@ -122,7 +133,11 @@ public Collection<ConsumerGroupListing> listGroups() {
      */
     public void deleteGroupIfExists(final String groupName) {
         if (this.exists(groupName)) {
-            this.deleteConsumerGroup(groupName);
+            try {
+                this.deleteConsumerGroup(groupName);
+            } catch (final GroupIdNotFoundException e) {
```

Comment:
This is very weird. When executing tests, sometimes exists returns true, presumably because Kafka Streams resetter has reset offsets before on an non-existent consumer group. But when calling delete, the consumer group does not exist anymore. When doing some logging on between, it works as well but that is not a very robust solution. In the end, I think catching this exception does not hurt